### PR TITLE
Update to graphql-java 4.2

### PR DESCRIPTION
**Pull request by tdraier**
No major changes, however we could have a v3 branch to continue support on graphql 3.0

### CODE REVIEW

File: src/test/java/graphql/annotations/GraphQLObjectTest.java [hunk 1]
```diff
@@ -622,7 +622,7 @@ public void queryOptional() {
         GraphQLObjectType object = GraphQLAnnotations.object(OptionalTest.class);
         GraphQLSchema schema = newSchema().query(object).build();
 
-        GraphQL graphQL = GraphQL.newGraphQL(schema).queryExecutionStrategy(new EnhancedExecutionStrategy()).build();
```
Comment (yarinvak) [inline src/test/java/graphql/annotations/GraphQLObjectTest.java:660]: Why did you remove the queryExecutionStrategy?

**Comment by yarinvak** [comment]
Looks like there are not many changes, it looks good.
By the way, version 5.0 of GraphQL-Java was released about a week ago, we should work on that too.

Can you write a little more detailed description about the parts you changed here?

File: src/test/java/graphql/annotations/GraphQLObjectTest.java (hunk 1 above)
Comment (Nitix) [reply src/test/java/graphql/annotations/GraphQLObjectTest.java:660]: He may removed it during development but forgot to put it back with its third commit. 

Also https://github.com/graphql-java/graphql-java-annotations/pull/105/files#diff-9599d6f1ee8ac402b47f46e351eead38L644 and https://github.com/graphql-java/graphql-java-annotations/pull/105/files#diff-9599d6f1ee8ac402b47f46e351eead38L675

Comment (tdraier) [reply src/test/java/graphql/annotations/GraphQLObjectTest.java:660]: Yes, I restored an EnhancedExecutionStrategy for gql4.2, but did not put it back on tests where it was not needed - it's only used for relay mutations, so kept it only in RelayTest. I could restore it in these tests also, but it won't change much.

Comment (yarinvak) [reply src/test/java/graphql/annotations/GraphQLObjectTest.java:660]: If its not necessary, no need

**Comment by tdraier** [comment]
Here's the list of changes :

- Changed dependency to graphql-java-4.2
- Updated EnhancedStrategyExecution to inherit from AsyncSerialExecutionStrategy, updated method signatures from ExecutionStrategy (now support Future)
- Removed TypeInfoWorkaround (method is now public)
- DataFetchingEnvironment constructor changed : added missing parameters when duplicating DataFetchingEnvironment
- Cannot pass null DataFetcher in builder anymore, use fake value for building relay mutation
- MethodDataFetcher : only arguments with values are now passed as parameters, use a map with names instead of assuming all arguments in sequence

File: src/main/java/graphql/annotations/EnhancedExecutionStrategy.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLObjectType;
+
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class EnhancedExecutionStrategy extends AsyncSerialExecutionStrategy {
+
+    private static final String CLIENT_MUTATION_ID = "clientMutationId";
```
Comment (DoctorVoid) [inline src/main/java/graphql/annotations/EnhancedExecutionStrategy.java]: why do you have this const?

Comment (tdraier) [reply src/main/java/graphql/annotations/EnhancedExecutionStrategy.java]: Sorry I don't understand the question ? The const was not introduced in this PR but in https://github.com/graphql-java/graphql-java-annotations/commit/f39d26f , and is still used in 3 places in the class.

**Comment by guy120494** [comment]
Can you please resolve the conflicts?

**Comment by tdraier** [comment]
Is it ok to merge ?
